packages/server/test/HeaderValidator.test.ts

```diff
@@ -0,0 +1,169 @@
+import {strict as assert} from 'node:assert'
+
+import {validateHeader} from '../src/validators/HeaderValidator'
+import {TUS_RESUMABLE} from '../src/constants'
+
+describe('HeaderValidator', () => {
+  describe('upload-offset', () => {
+    it('should validate a number', (done) => {
+      const value = '1234'
+      assert.equal(validateHeader('upload-offset', value), true)
+      done()
+    })
+
+    it('should invalidate a negative number', (done) => {
+      const value = '-4'
+      assert.equal(validateHeader('upload-offset', value), false)
+      done()
+    })
+
+    it('should invalidate a non number', (done) => {
+      assert.equal(validateHeader('upload-length', 'hello'), false)
+      assert.equal(validateHeader('upload-length', '0100'), false)
+      assert.equal(validateHeader('upload-length', '0asd100'), false)
+      assert.equal(validateHeader('upload-length', '1asd100'), false)
+      done()
+    })
+  })
+
+  describe('upload-length', () => {
+    it('should validate a number', (done) => {
+      const value = '1234'
+      assert.equal(validateHeader('upload-length', value), true)
+      done()
+    })
+
+    it('should invalidate a number < 1', (done) => {
+      assert.equal(validateHeader('upload-length', '-1'), false)
+      done()
+    })
+
+    it('should invalidate a non number', (done) => {
+      assert.equal(validateHeader('upload-length', 'hello'), false)
+      assert.equal(validateHeader('upload-length', '0100'), false)
+      assert.equal(validateHeader('upload-length', '0asd100'), false)
+      assert.equal(validateHeader('upload-length', '1asd100'), false)
+      done()
+    })
+  })
+
+  describe('upload-defer-length', () => {
+    it('should validate 1', (done) => {
+      const value = '1'
+      assert.equal(validateHeader('upload-defer-length', value), true)
+      done()
+    })
+
+    it('should invalidate a number !== 1', (done) => {
+      assert.equal(validateHeader('upload-defer-length', '0'), false)
+      assert.equal(validateHeader('upload-defer-length', '1234'), false)
+      assert.equal(validateHeader('upload-defer-length', '-1'), false)
```

Comment:
Added some more tests that might fail if `Number.parseInt(value) === 1` is used.

```suggestion
      assert.equal(validateHeader('upload-defer-length', '-1'), false)
      assert.equal(validateHeader('upload-defer-length', '+1'), false)
      assert.equal(validateHeader('upload-defer-length', ' 1 '), false) // test leading and trailing whitespaces
```